Code/Tools/AssetProcessor/native/AssetManager/SourceAssetReference.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <QString>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/IO/Path/Path.h>
+#include <utilities/IPathConversion.h>
+#include <native/AssetManager/assetScanFolderInfo.h>
+
+namespace AssetProcessor
+{
+    class SourceAssetReference
```

Comment:
Nit: It may worth having some comments for explaining the responsibility of this class. I'm also wondering whether we want to move this to AzFramework or AzToolFramework since other content tools may also reference source assets via this structure going forward.